src/bufr/BufrParser/Query/DataProvider.cpp

```diff
@@ -98,5 +101,60 @@ namespace bufr {
     {
         delete_table_data_f();
     }
+
+    TypeInfo DataProvider::getTypeInfo(FortranIdx idx) const
+    {
+        static const unsigned int UNIT_STR_LEN = 24;
+        static const unsigned int DESC_STR_LEN = 55;
```

Comment:
Is 55 a hard-wired limit?  Kinda arbitrary.  This may be entirely unimportant, but I often like to declare string lengths to remain multiples of 8 for memory reasons; so could this be 64?  No idea what downstream effects would occur.